Mathlib/CategoryTheory/Enriched/FunctorCategory.lean

```diff
@@ -216,4 +216,126 @@ noncomputable def enrichedOrdinaryCategory [∀ (F₁ F₂ : J ⥤ C), HasEnrich
   homEquiv_id _ := homEquiv_id V _
   homEquiv_comp f g := homEquiv_comp V f g
 
+variable {J C}
+
+section
+
+variable (G : K ⥤ J) [HasEnrichedHom V F₁ F₂]
+  [HasEnrichedHom V (G ⋙ F₁) (G ⋙ F₂)]
+
+/-- If `F₁` and `F₂` are functors `J ⥤ C`, and `G : K ⥤ J`,
+then this is the induced morphism
+`enrichedHom V F₁ F₂ ⟶ enrichedHom V (G ⋙ F₁) (G ⋙ F₂)` in `V`
+when `C` is a category enriched in `V`. -/
+noncomputable def precompEnrichedHom :
+    enrichedHom V F₁ F₂ ⟶ enrichedHom V (G ⋙ F₁) (G ⋙ F₂) :=
+  end_.lift (fun _ ↦ enrichedHomπ _ _ _ _)
+    (fun _ _ f ↦ enrichedHom_condition V F₁ F₂ (G.map f))
+
+@[reassoc (attr := simp)]
+lemma precompEnrichedHom_π (k : K) :
+    precompEnrichedHom V F₁ F₂ G ≫ enrichedHomπ _ _ _ k = enrichedHomπ _ _ _ (G.obj k) := by
+  apply end_.lift_π
+
+end
+
+section
+
+/-- Given functors `F₁` and `F₂` in `J ⥤ C`, where `C` is a category enriched in `V`,
+this condition allows the definition of `functorEnrichedHom V F₁ F₂ : J ⥤ V`. -/
+abbrev HasFunctorEnrichedHom :=
+  ∀ (j : J), HasEnrichedHom V (Under.forget j ⋙ F₁) (Under.forget j ⋙ F₂)
+
+variable [HasFunctorEnrichedHom V F₁ F₂]
+
+instance {j j' : J} (f : j ⟶ j') :
+    HasEnrichedHom V (Under.map f ⋙ Under.forget j ⋙ F₁)
+      (Under.map f ⋙ Under.forget j ⋙ F₂) :=
+  inferInstanceAs (HasEnrichedHom V (Under.forget j' ⋙ F₁) (Under.forget j' ⋙ F₂))
+
+/-- Given functors `F₁` and `F₂` in `J ⥤ C`, where `C` is a category enriched in `V`,
+this is the enriched hom functor from `F₁` to `F₂` in `J ⥤ V`. -/
+@[simps!]
+noncomputable def functorEnrichedHom : J ⥤ V where
+  obj j := enrichedHom V (Under.forget j ⋙ F₁) (Under.forget j ⋙ F₂)
+  map f := precompEnrichedHom V (Under.forget _ ⋙ F₁) (Under.forget _ ⋙ F₂) (Under.map f)
+  map_id X := by
+    dsimp
+    ext j
+    rw [precompEnrichedHom_π, id_comp]
+    congr 1
+    simp [Under.map, Comma.mapLeft]
+    rfl
```

Comment:
```suggestion
    dsimp
    ext j
    simp only [diagram_obj_obj, Functor.comp_obj, Under.forget_obj, Under.map_obj_right,
      end_.lift_π, id_comp]
    congr
    simp [Under.map, Comma.mapLeft]
    rfl
```

Automation seems to not be working well enough in this file. The above change gets rid of an `erw` if `precompEnrichedHom` is changed to an `abbrev`, but it doesn't solve the other `erw`s. I think `precompEnrichedHom_π` is a bad simp lemma, but changing `enrichedHomπ` (which is an `abbrev`) in it to `end_.π` is no better. 

Otherwise LGTM

Reply:
Thanks. I have removed `precompEnrichedHom_π` by making `precompEnrichedHom` an abbrev.
As the internals of some terms are `dsimp`ed, I do not see any good way to remove the `erw`.